datastore/fertiscan/__init__.py

```diff
@@ -69,3 +71,38 @@ async def register_analysis(
     except Exception as e:
         print(e.__str__())
         raise Exception("Datastore unhandeled error")
+
+
+async def update_inspection(
+    cursor: Cursor,
+    inspection_id: str | UUID,
+    user_id: str | UUID,
+    updated_data: dict | data_inspection.Inspection,
+):
+    """
+    Update an existing inspection record in the database.
+
+    Parameters:
+    - cursor (Cursor): Database cursor for executing queries.
+    - inspection_id (str | UUID): UUID of the inspection to update.
+    - user_id (str | UUID): UUID of the user performing the update.
+    - updated_data (dict | data_inspection.Inspection): Dictionary or Inspection model containing updated inspection data.
+
+    Returns:
+    - data_inspection.Inspection: Updated inspection data from the database.
+
+    Raises:
+    - InspectionUpdateError: If an error occurs during the update.
+    """
+    if isinstance(inspection_id, str):
+        inspection_id = UUID(inspection_id)
+    if isinstance(user_id, str):
+        user_id = UUID(user_id)
```

Comment:
I also like to check is_user_register() before linking a uuid so there are no DB error. 

Do we want restrict users on which inspection they can validate?

Reply:
Done. Yes.